### DISCUSSION TITLE
Document clone via API

### DISCUSSION THREAD

**Discussion by riccardominato** [ravendb/ravendb#18316] [Q&A] [answered]
I need to edit some document IDs from a collection whose documents have time series.

I know this is not possible in an "edit the ID" way and I should clone every document giving to the clones the ID I desire.
This is achievable using the "Clone" button from RavenDB Studio.

I was wondering if there is an API that obtains the same result and can be called programmatically.

I tried with [patch clone](https://ravendb.net/docs/article-page/6.0/csharp/client-api/operations/patching/single-document#clone-document), but it doesn't copy the time series.

Thank you.

**Comment by ayende** [comment, marked as answer]
That is actually done as a few separate commands, not a single entry.
That said, you can merge all of them to a single transaction.

The API for that is `CopyTimeSeriesCommandData`, `CopyAttachmentCommandData`, etc.
https://github.com/ravendb/ravendb/blob/971172305c4641461b1472ddafa4b586e4e27d2b/test/SlowTests/Client/TimeSeries/Issues/RavenDB_14386.cs#L23

**Comment by riccardominato** [comment]
@ayende Thank you very much for the quick response! Your suggestion put me on the right track.

Just a note: you may want to make `CopyTimeSeriesCommandData` public. I had to copy the source code since currently it's marked as `internal`. Not a big deal, but since `CopyAttachmentCommandData` is public i would say also `CopyTimeSeriesCommandData` could be.
